internal/metrics/client.go

```diff
@@ -9,20 +9,16 @@ import (
 	"net/http"
 	"net/url"
 	"strconv"
+	"strings"
 	"time"
 
 	"rudder-load/internal/parser"
 )
 
-type MimirClient interface {
-	Query(ctx context.Context, query string, time int64) (QueryResponse, error)
-	QueryRange(ctx context.Context, query string, start int64, end int64, step string) (QueryResponse, error)
-	GetMetrics(ctx context.Context, mts []parser.Metric) ([]MetricsResponse, error)
-}
-
-type mimirClient struct {
+type MetricsClient struct {
```

Comment:
[optional] We could rename this to `Client` given that when it'll be used via its package name anyway i.e. `metrics.Client` it should be pretty clear what it does anyway. Also it could be a `Fetcher` instead of `Client`.

Reply:
Simplified the structure name to Fetcher. Left the constructor names as NewMetricsFetcher and NewLocalMetricsFetcher for more clarity when used.